lib/BC/PcodeCFG.cpp

```diff
@@ -0,0 +1,166 @@
+#include <remill/BC/PCodeCFG.h>
+
+#include <algorithm>
+#include <cstddef>
+#include <sleigh/op.hh>
+#include <sleigh/opcodes.hh>
+#include <sleigh/pcoderaw.hh>
+#include <map>
+#include <variant>
+#include <vector>
+
+#include "lib/Arch/Sleigh/ControlFlowStructuring.h"
+
+namespace remill {
+namespace sleigh {
+
+PcodeCFG
+PcodeCFGBuilder::CreateCFG(const std::vector<RemillPcodeOp> &linear_ops) {
+  return PcodeCFGBuilder(linear_ops).Build();
+}
+
+
+std::vector<size_t> PcodeCFGBuilder::GetBlockStarts() {
+  std::vector<size_t> res;
+  res.push_back(0);
+
+
+  for (size_t curr_index = 0; curr_index < this->linear_ops.size();
+       curr_index += 1) {
+    auto interproc = this->GetIntraProcTargets(curr_index);
+    res.insert(res.end(), interproc.begin(), interproc.end());
+  }
+  return res;
+}
+
+PcodeBlock PcodeCFGBuilder::BuildBlock(size_t start_ind, size_t next_start) {
+  std::vector<RemillPcodeOp> ops;
+
+  for (size_t i = start_ind; i < next_start; i++) {
+    ops.push_back(this->linear_ops[i]);
+  }
+
```

Comment:
```
using pcode_ops_t = ...;

auto begin = std::next( linear_ops.begin() + start );
auto end = std::next( linear_ops.begin() + next_start );
return ...( start, pcode_ops_t( begin, end ), ... );
```

Reply:
[fa650f5](https://github.com/lifting-bits/remill/pull/645/commits/fa650f549045057c1767379fd45c5351fe6a9220)

I decided to just use `std::copy` here.